lib/internal/modules/esm/public_loader_proxy.js

```diff
@@ -0,0 +1,42 @@
+const { Worker } = require('internal/worker');
+
+
+/**
+ * The communication vehical constituting all memory shared between the main and the worker
+ */
+const comsChannel = new SharedArrayBuffer(2048);
+/**
+ * The lock/unlock segment of the shared memory. Atomics require this to be a Int32Array. This
+ * segment is used to tell the main to sleep when the worker is worker, and vice verse (for the
+ * worker to sleep whilst the main is working).
+ */
+const lock = new Int32Array(comsChannel, 0, 4);
+/**
+ * The request & response segment of the shared memory. TextEncoder/Decoder (needed to convert
+ * requests & responses into a format supported by the coms channel) reads and writes with
+ * Uint8Array.
+ */
+const requestResponseData = new Uint8Array(comsChannel, 4, 2044);
+
+const worker = new Worker('./worker.mjs', {
+  workerData: { comsChannel },
+});
+worker.unref(); // ! Allows the process to eventually exit when worker is in its final sleep.
+
+Atomics.wait(lock, 0, 1); // ! Block this module until the worker is ready.
+
+function makeRequest(type, data) {
+  requestResponseData.fill(0); // Erase any previous request/response (it's already been handled)
+  const request = JSON.stringify({ data, type });
```

Comment:
Do we need `__proto__: null` here?

Reply:
It's probably not a big deal, but if there's enumerable prototype pollution, wouldn't it get serialised and bloat the data?

Reply:
I have no idea. I was honestly asking a question 😄 

@aduh95 is there a guide to `__proto__` somewhere? Like we have for primordials.

Reply:
Nah `JSON.stringify` only cares about own (enumerable) properties (step 6.a in https://tc39.es/ecma262/#sec-serializejsonobject), so prototype pollution is not a concern here.

Reply:
Should we use `v8.serialize` instead? It outputs a `Uint8Array` directly, it would save use the `TextEncoder` and `TextDecoder` instantiation.

Reply:
> Should we use `v8.serialize` instead? It outputs a `Uint8Array` directly, it would save use the `TextEncoder` and `TextDecoder` instantiation.

That's quite a trick. Something to note in a comment.

Reply:
Huzzah! This addressed a few things 🙌

* it's simpler / easier to grok
* runtime in my experiment decreased from `0m2.111s` → `0m2.105s` <sup>[1]</sup>
* the `.replaceAll('\x00', '')` is not needed anymore

1. The minimum-possible runtime of the experiment is 2s, so 0.105s for node to start up and run the experiment seems pretty good!